pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -247,12 +256,19 @@ func computeActiveCondition(ctx context.Context, pa *pav1alpha1.PodAutoscaler, p
 		if pc.want > 0 || !pa.Status.IsInactive() {
 			pa.Status.MarkActivating(
 				"Queued", "Requests to the target are being buffered as resources are provisioned.")
+		} else {
+			// This is for the initialScale 0 case. In the first iteration, minReady is 0,
+			// but for the following iterations, minReady is 1. pc.want will continue being
+			// -1 until we start receiving metrics, so we will end up here.
+			// Even though PA is already been marked as inactive in the first iteration, we
+			// still need to set it again. Otherwise reconciliation will fail with NewObservedGenFailure
+			// because we cannot go through one iteration of reconciliation without setting
+			// some status.
+			pa.Status.MarkInactive("NoTraffic", "The target is not receiving traffic.")
```

Comment:
The way I read it, this is redundant.
On first iteration we'll enter the first switch case and mark PA as **Inactive**.
On the next ones we'll enter here (0 < 1). But as you mentioned `pc.want=-1` and the `pa.Status==Inactive`: thus the `if` above will always evaluate to false (unless we receive requests and positive metrics and `pc.want` becomes positive). Thus you're just marking `Inactive` with `Inactive` again.

Reply:
Correct, we are marking Inactive with Inactive again here. This is still needed because we will be getting the NewObservedGenFailure during reconciliation post processing, because we cannot go through one iteration of reconciliation without setting some sort of status.

Reply:
Hm, so just updating the status with the same value updates ObsGen? 
@whaught, Weston, if change in inputs didn't yield any change in status, why would this be an error?

Reply:
@whaught i'm not sure if my understanding is correct. I think it's because PA has been updated during the reconciliation, therefore there's a difference in ObsGen and Gen, but the status is not updated, which causes the failure: https://github.com/knative/pkg/blob/deb6b33d2a6c114f596f52630e85c475bf43abce/reconciler/reconcile_common.go#L50-L53

Reply:
When Spec changes, we reset Ready to unknown with a dummy message that the reconciler didn't set anything for a new spec before calling ReconcileKind. The reconciler is expected to set _something_ upon observation of a new generation (or is left with the default message as a warning)

Reply:
I wonder the spec change in PA is 🤔 
But anyway, it's interesting, since this does not change `ready` (before and after would be `unknown`).